presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -1750,9 +1750,24 @@ private static Number toNumber(Object value)
         if (value instanceof Boolean) {
             return ((Boolean) value) ? 1 : 0;
         }
+        if (value instanceof String) {
+            return convertStrToBigDecimal((String) value);
+        }
         throw new SQLException("Value is not a number: " + value.getClass().getCanonicalName());
     }
 
+    private static BigDecimal convertStrToBigDecimal(String value) throws SQLException
```

Comment:
We avoid abbreviations like "str". However, here you can just call the method `toBigDecimal`

Reply:
fixed